lib/rust/ensogl/core/src/display/scene/layer.rs

```diff
@@ -918,16 +967,35 @@ impl SublayersModel {
     /// by calling the `update` method on [`Group`], which usually happens once per animation
     /// frame.
     pub fn all(&self) -> Vec<Layer> {
-        self.layers.iter().filter_map(|t| t.upgrade()).collect()
+        self.iter_all().collect()
+    }
+
+    /// Iterator of all layers, ordered according to the defined depth-order dependencies. It will
+    /// borrow the sublayers for as long as the iterator is alive.
```

Comment:
I dont see any borrowing in the code

Reply:
Right. The borrow is directly on `&self` here, which function signature makes unambiguous by `'_` lifetime on returned iterator. So that extra comment is not needed here.